eduNEXT/frontend-app-ora-grading
# feat: open response assesment detail (problem steps) UI

> ref: https://github.com/openedx/frontend-app-ora-grading/pull/263

## Description

This PR allows us to see in detail each step of an open response, which means **Training**, **Peer**, **Self**, and **Staff**, and can also see feedback received and feedback given.

![demo](https://github.com/eduNEXT/frontend-app-ora-grading/assets/64033729/c9294c00-ba82-41c8-9cb9-968d38e02c39)

### Responses List

| Before | Now |
|--------|--------|
| ![image](https://github.com/eduNEXT/frontend-app-ora-grading/assets/64033729/34929ee7-2fc1-4432-ae19-3f02acccdc45) | ![image](https://github.com/eduNEXT/frontend-app-ora-grading/assets/64033729/8d7937bb-dd72-4d85-9a8c-4450a0d99412) | 

### Assessments Feedback Received
![image](https://github.com/eduNEXT/frontend-app-ora-grading/assets/64033729/986579f8-6a53-4870-9619-abf0b69bc9d2)

### Assessments Feedback Given
![image](https://github.com/eduNEXT/frontend-app-ora-grading/assets/64033729/695dae22-2307-4047-98ac-a53fdaf59bf2)

Here's what you see in the video explained:

- There is an ORA open response type **Staff Assessment Only**.
- The instructor opens the new experience of ORA that contains the submissions of that Open response.
- The instructor can see the details and more information, such as email, full name, list of responses, and feedback in detail.

**Why are we proposing these changes?**

This improvement enables instructors to access detailed information for each Open Response, making it easier for them to review ORA responses and maintain better control over the feedback process.

## Dependencies
- edx-platform: https://github.com/openedx/edx-platform/pull/33632 **(Merged)**
- edx-ora2: https://github.com/openedx/edx-ora2/pull/2101 **(Merged)**

## Testing Instructions

Using Tutor:
1. Clone this repository in your environment and checkout to this branch.
2. Into the repo run `nvm use && npm install`
3. Create a mount of this MFE with `tutor mounts add frontend-app-course-authoring`
4. Run `tutor dev start -d` to create the new container of ORA Grading.
5. From Studio create a unit with an open response type **Staff Assessment Only**.
6. From LMS submit the response.
7. Go to the **Instructor Tab** > **Open Responses**.
8. In the unit click on **View and grade responses**. Here you can see the new fields added to the table (Email, Problem steps, and Action)
9. Click in **View details**, and look at the new assessment tables.
